### PR TITLE
feat(tests): Implemented tests for 100% coverage of pinned_post_screen

### LINKED ISSUE #2903: Improve test code coverage for lib/views/after_auth_screens/feed/pinned_post_screen.dart
We will need unittests done for all methods, classes and functions found in this file.

Any widgets referenced in this file must also have unittests done

### PR Acceptance Criteria

1. When complete this file must show **100%** coverage when merged into the code base.
1. [The current code coverage for the file can be found here](https://app.codecov.io/gh/PalisadoesFoundation/talawa/tree/develop/?displayType=list)
1. CodeRabbit must approve all your changes to the test file(s) and the file(s) under test
1. The PR will show a report for the code coverage for the file you have added. You can use that as a guide
1. No functionality must be removed to achieve this goal
1. All tests must be valid and truly test the code base

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Prakharpan-dey**
### What kind of change does this PR introduce?

feat(tests): Implemented tests for pinned_post_screen

### Issue Number:

Fixes #2903 

### Did you add tests for your changes?

Yes

- [x] Tests are written for all changes made in this PR.
- [x] Test coverage meets or exceeds the current coverage (~90/95%).

Snapshots/Videos:

image

### If relevant, did you update the documentation?

Not relevant

### Summary

Implemented tests for 100% coverage of pinned_post_screen

### Does this PR introduce a breaking change?

No
### Checklist for Repository Standards
- [x] Have you reviewed and implemented all applicable `coderaabbitai` review suggestions?
- [x] Have you ensured that the PR aligns with the repository’s contribution guidelines?

### Other information

None

### Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?
Yes

## Summary by CodeRabbit

* **Tests**
  * Expanded pinned-post test coverage with scenarios for image load failures, empty attachments, and null/empty captions.
  * Test harness made configurable so tests can supply different captions, attachment sets, and cache behaviors to exercise edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: test/views/after_auth_screens/feed/pinned_post_screen_test.dart [hunk 1]
```diff
@@ -129,4 +145,39 @@ void main() {
...
+  testWidgets('Check if empty attachments are handled gracefully',
+      (tester) async {
+    await mockNetworkImagesFor(() async {
+      await showPinnedPostScreen(
+        tester,
+        attachments: [],
+      );
+      expect(find.text('Sample Title'), findsOneWidget);
+    });
+  });
```
[the remark this answers is not in the document]
Comment (Prakharpan-dey) [reply test/views/after_auth_screens/feed/pinned_post_screen_test.dart:181]: CachedNetworkImage is always rendered in the Column's children list.

**Review by palisadoes — APPROVED** [review]
